lib/active_job/queue_adapters/sidekiq_publisher_adapter.rb

```diff
@@ -36,6 +45,10 @@ def job_attributes(job)
           args: [job.serialize],
         }
       end
+
+      def sidekiq_adapter
+        @_sidekiq_adapter ||= ActiveJob::QueueAdapters::SidekiqAdapter.new
+      end
```

Comment:
❓ This will still pick up any application-specific configuration for Sidekiq I'm guessing right?

Reply:
I was concerned about that too, and thought of a few ways of doing this direct enqueuing.

I wanted to stay as high-level as possible rather than calling lower-level methods in Sidekiq in order to enqueue. I saw that [this](https://github.com/rails/rails/blob/774c630068e69547fabd6bddedd62a712ae7e736/activejob/lib/active_job/queue_adapter.rb#L51) is where we, in Rails, assign which adapter to use, and it just instantiates the adapter with no args. So it *felt* safe to instantiate our own and call [enqueue](https://github.com/rails/rails/blob/316338acf76c34b529e7d883a916ace798ad8efa/activejob/lib/active_job/queue_adapters/sidekiq_adapter.rb#L21) or [enqueue_at](https://github.com/rails/rails/blob/316338acf76c34b529e7d883a916ace798ad8efa/activejob/lib/active_job/queue_adapters/sidekiq_adapter.rb#L28).

That said, I did test that it still runs client and server middleware. What else are you thinking?

Reply:
Looking at the Sidekiq adapter more it doesn't seem like there's much/any configuration for it so it should just be using the application configuration already. Just wanted to double check.